### PR TITLE
Make android modal to appear underneath translucent StatusBar

### PR DESCRIPTION
**Pull request by pfulop**
## Summary

As described in #26559 , the modal is not appearing underneath translucent StatusBar so you can see previous view underneath the StatusBar instead. As a solution you can now provide prop to set the modal to have translucent StatusBar and therefore the content will appear underneath. I tried to reuse layout flags that are possibly set if you use StatusBar component but sadly values in them do not reflect the props set by StatusBar component.

## Changelog

[Android] [added] - Added statusBarTranslucent prop to Modal component, to indicate if StatusBar should appear translucent.

### With StatusBar translucent

image

``` <Modal statusBarTranslucent>```
### Without

image

``` <Modal>```

### CODE REVIEW

**Comment by dulmandakh** [comment]
IIRC, you can use StatusBar to achieve the same

**Comment by pfulop** [comment]
@dulmandakh unfortunately no. There are two reasons: 
1.  frameLayout.setFitsSystemWindows(true); is acting strangely with frame layour (which is 1st child layout in the modal )
2. Window flags seems to be cleared whenever the modal is shown.

**Comment by pfulop** [comment]
@mdvacca  Yes I am planning to add it to the docs. Should I open PR already or wait for this to be merged?

File: Libraries/Modal/Modal.js [hunk 1]
```diff
@@ -69,6 +69,14 @@ export type Props = $ReadOnly<{|
    */
   transparent?: ?boolean,
 
+  /**
+   * The `statusBarTranslucent` prop determines whether your modal should go under
+   * the system statusbar.
```
[the remark this answers is not in the document]
Comment (nicolasburtey) [reply Libraries/Modal/Modal.js]: seems comment has not been added :)

**Pull request closed without merging**